manifest-build-tools/application/reprove.py

```diff
@@ -0,0 +1,243 @@
+            self._manifest.validate_manifest()
+        except KeyError as error:
+            print "Failed to create a Manifest instance for the manifest file {0} \nERROR:\n{1}"\
+                  .format(manifest_path, error.message)
+            sys.exit(1)
+         
+        for repo in self._manifest.get_repositories():
+            repo['directory-name'] = self.directory_for_repo(repo)
+
+
+    def check_builddir(self):
+        """
+        Checks the given builddir name and force flag. Deletes exists directory if one already
+        exists and --force is set
+        :return: None
+        """
+        if os.path.exists(self._builddir):
+            if self._force:
+                shutil.rmtree(self._builddir)
+                print "Removing existing data at {0}".format(self._builddir)
+            else:
+                print "Unwilling to overwrite destination builddir of {0}".format(self._builddir)
+                sys.exit(1)
+
+        os.makedirs(self._builddir)
+
+        
+    def get_repositories(self):
+        """
+        Issues checkout commands to dictionaries within a provided manifest
+        :return: None
+        """
+        repo_list = self._manifest.get_repositories()
+        try:
+            self.repo_operator.clone_repo_list(repo_list, self._builddir, jobs=self._jobs)
+        except RuntimeError as error:
+            print "Exiting due to error: {0}".format(error)
+            sys.exit(1)
+            
+
+    def directory_for_repo(self, repo):
+        if 'checked-out-directory-name' in repo:
+            repo_directory = repo['checked-out-directory-name']
+        else:
+            if 'repository' in repo:
+                repo_url = repo['repository']
+                repo_directory = strip_suffix(os.path.basename(repo_url), ".git")
+            else:
+                raise ValueError("no way to find basename")
+
+        repo_directory = os.path.join(self._builddir, repo_directory)
+        return repo_directory
+
+
+
+def parse_command_line(args):
+    """
+    Parse script arguments.
+    :return: Parsed args for assignment
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--manifest",
+                        required=True,
+                        help="repository manifest file",
+                        action="store")
+    parser.add_argument("--builddir",
+                        required=True,
+                        help="destination for checked out repositories",
+                        action="store")
+    parser.add_argument("--force",
+                        help="use destination dir, even if it exists",
+                        action="store_true")
+    parser.add_argument("--git-credential",
+                        help="Git credentials for CI services",
+                        action="append")
+    parser.add_argument("--jobs",
+                        default=1,
+                        help="Number of parallel jobs to run",
```

Comment:
What's the recommendation to determine the "jobs" number? do you want to explain in code ?